src/activities/discussions/DiscussionTopicEntity.js

```diff
@@ -21,16 +21,58 @@ export class DiscussionTopicEntity extends Entity {
 	}
 
 	/**
-	 * Updates the discussion topic to have the given name
-	 * @param {string} name Name to set on the discussion topic
+	 * @summary Checks if topic entity has changed, primarily used for dirty check
+	 * @param {object} topic the topic that's being modified
 	 */
-	async setName(name) {
-		const action = this.canEditName() && this._entity.getActionByName(Actions.discussions.topic.updateName);
-		if (!action) {
-			return;
+	equals(topic) {
+		const diffs = [
+			[topic.name, this.name()],
+		];
+
+		for (const [current, initial] of diffs) {
+			if (current !== initial) {
+				return false;
+			}
 		}
 
+		return true;
+	}
+
+	/**
+	 * @summary Fires all the formatted siren actions collectively
+	 * @param {object} topic the topic that's being modified
+	 */
+	async save(topic) {
+		if (!topic) return;
+
+		const updateNameAction = this._formatUpdateNameAction(topic);
+
+		const sirenActions = [
+			updateNameAction,
+		];
+
+		await performSirenActions(this._token, sirenActions);
+	}
+
+	/**
+	 * @summary Formats action and fields if topic name has changed and user has edit permission
+	 * @param {object} topic the topic that's being modified
+	 * @returns {object} the appropriate action/fields to update
+	 */
+	_formatUpdateNameAction(topic) {
```

Comment:
I found it hard to read the quiz entity when the action was at the bottom of the file, away from the property (name) getter. Maybe we can follow the style of the assignment entity which kept them close?
https://github.com/BrightspaceHypermediaComponents/siren-sdk/blob/master/src/activities/assignments/AssignmentEntity.js#L17-L32